src/devices/Ad7193/Ad7193.cs

```diff
@@ -0,0 +1,842 @@
+﻿using System;
+using System.Collections.Concurrent;
+using System.Collections.Generic;
+using System.Device.Spi;
+using System.Diagnostics;
+using System.Text;
+using System.Threading;
+using System.Threading.Tasks;
+using Ad7193.Metadata;
+using Iot.Units;
+
+namespace Iot.Device.Ad7193
+{
+    /// <summary>
+    /// Represents the Analog Devices AD7193, the 4-channel, 4.8 kHz, ultralow noise, 24-bit sigma-delta ADC with PGA
+    /// </summary>
+    public class Ad7193 : IDisposable
+    {
+        private readonly object _spiTransferLock = new object();
+        private readonly Stopwatch _stopWatch = new Stopwatch();
+
+        private SpiDevice _spiDevice = null;
+
+        /// <summary>
+        /// Metadata of AD7193
+        /// </summary>
+        protected static Ad7193Metadata metadata = new Ad7193Metadata();
+
+        /// <summary>
+        /// The list of received ADC values
+        /// </summary>
+        private readonly BlockingCollection<AdcValue> _adcValues = new BlockingCollection<AdcValue>();
+
+        /// <summary>
+        /// The event that is fired every time a new value is received from the ADC
+        /// </summary>
+        public event EventHandler<AdcValueReceivedEventArgs> OnValueReceived;
+
+        private readonly StringBuilder _sb = new StringBuilder();
+
+        private readonly uint[] _registerCache = { 0x00, 0x080060, 0x000117, 0x000000, 0xa2, 0x00, 0x000000, 0x000000 };
+        private readonly byte[] _registerSize = { 1, 3, 3, 3, 1, 1, 3, 3 };
+
+        /// <summary>
+        /// The external reference voltage value. The default is 2.5V on REFIN1+ and REFIN1- (on the Digilent Pmod AD5 board)
+        /// </summary>
+        public double VoltageReference { get; set; } = 2.50f;
+
+        /// <summary>
+        /// Polarity select bit: true is unipolar, false is bipolar operation.
+        /// </summary>
+        public bool Unipolar
```

Comment:
it would be preferred to create an enum `Polarity` with both values although bool is alright too